### PR TITLE
Allow SSL verification when using a proxy

### LINKED ISSUE #460: Proxy settings Graph API turn off SSL verification
In our production environment, all outbound HTTP requests have to be sent via our internal proxy for security reasons.

If we apply our proxy settings, when `\Microsoft\Graph\Http\GraphRequest::createGuzzleClient` is run the SSL verification is turned off:

```php
    protected function createGuzzleClient()
    {
        $clientSettings = [
            'base_uri' => $this->baseUrl,
            'http_errors' => $this->http_errors,
            'headers' => $this->headers
        ];
        if ($this->proxyPort !== null) {
            $clientSettings['verify'] = false;
            $clientSettings['proxy'] = $this->proxyPort;
        }
        $client = new Client($clientSettings);

        return $client;
    }
```

I can see that the normal usage would be adding Fiddler or similar which _would_ require the SSL verification to be turned off (because you're intentionally adding a MITM), but in our case the proxy won't cause an SSL failure.

Would you accept a PR to make it so that we could turn SSL verification on? I was thinking something like:

 * `\Microsoft\Graph\Graph::setProxyPort` should have an additional param for SSL verification (defaulting to false to prevent a BC break)
 * This gets passed to the `GraphCollectionRequest` and `GraphRequest` constructors, and `\Microsoft\Graph\Http\GraphRequest::createGuzzleClient` uses it 
[AB#9228](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9228)

### PR DESCRIPTION
**Pull request by PatrickRose**
In some cases (ie enterprise environments), a user might need to use a
proxy to make the outbound requests. This allows those users to use a
proxy but *without* turning off SSL verification

Fixes #460

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/msgraph-sdk-php/pull/476)

### CODE REVIEW

File: src/Http/GraphRequest.php [hunk 1]
```diff
@@ -134,6 +135,7 @@ public function __construct($requestType, $endpoint, $accessToken, $baseUrl, $ap
         $this->timeout = 100;
         $this->headers = $this->_getDefaultHeaders();
         $this->proxyPort = $proxyPort;
+        $this->proxyVerifySSL = $proxyVerifySSL;
```
[the remark this answers is not in the document]
Comment (PatrickRose) [reply src/Http/GraphRequest.php:144]: Added - thanks!

[the remark this answers is not in the document]
Comment (guilliamxavier) [reply src/Http/GraphRequest.php:144]: Confirmed. But it seems that I can't resolve/dismiss my own review...

[the remark this answers is not in the document]
Comment (baywet) [reply src/Http/GraphRequest.php:144]: Resolving for you :)

**Review by Ndiritu — APPROVED** [review]
Thank you for your contribution @PatrickRose

**Review by MIchaelMainer — APPROVED** [review]

**Comment by PatrickRose** [comment]
Thanks for merging :)

Is there a rough date for when the next release would be? We've applied the patch locally but be useful to have a rough date for when we can just take the latest version.

**Comment by Ndiritu** [comment]
You're welcome @PatrickRose. I've just released the changes under `1.32.0`

**Pull request merged**